### DISCUSSION TITLE
Index size and timestamps

### DISCUSSION THREAD

**Discussion by oiBio** [ravendb/ravendb#15389] [General]
Hello,

we have an index with multiple stored DateTime fields and i was wondering how we could reduce the size of the index.
Regarding the timestamps in the table we do not care about the milliseconds or seconds. 

So one of the first ideas was, to cut these parts of the timestamps inside the index. 
We do this like this `TimeStamp =  vc.TimeStamp.AddTicks(-vc.TimeStamp.Ticks % TimeSpan.TicksPerSecond),`

Does this have an impact on the actual index size or is there a better idea how to do that?

**Comment by georgiosd** [comment]
You can convert to ISO string and cut out what you dont need. I would
assume that will reduce the index size.


On Tue, 6 Dec 2022 at 11:41, Tobias Bartsch ***@***.***>
wrote:

> Hello,
>
> we have an index with multiple stored DateTime fields and i was wondering
> how we could reduce the size of the index.
> Regarding the timestamps in the table we do not care about the
> milliseconds or seconds.
>
> So one of the first ideas was, to cut these parts of the timestamps inside
> the index.
> We do this like this TimeStamp =
> vc.TimeStamp.AddTicks(-vc.TimeStamp.Ticks % TimeSpan.TicksPerSecond),
>
> Does this have an impact on the actual index size or is there a better
> idea how to do that?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ravendb/ravendb/discussions/15389>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMYG5V72H27YQ2XJEA5RQ3WL4C3XANCNFSM6AAAAAASVJBVSA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Georgios Diamantopoulos
Founder
Zero to MVP
https://zerotomvp.dev

**Reply by garayx** [reply]
you might want to use the datetime ticks, which will save the date as long I think it will reduce the term size as well.

**Comment by ayende** [comment]
What is the size of your index right now?
In general, normalizing the dates to a minute / hour granularity is a great thing, since it reduces the number of terms we use.
Note that you want to keep it as a `DateTime` and not do string concat (we have optimizations on date ranges).

**Reply by oiBio** [reply]
Hi currently its ~17gb and indexes about 2.500.000 documents. mostly ids and timestamps 

So normalizing does/can have an impact on the size
